meta/src/utils.rs

```diff
@@ -0,0 +1,32 @@
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::{Attribute, Field, Ident, Meta, PathSegment, Type};
+
+// https://users.rust-lang.org/t/how-to-use-a-vector-of-tokenstreams-created-with-quote-within-quote/81092
+pub fn make_block(name: &Ident, functions: &Vec<TokenStream2>) -> TokenStream2 {
+    quote! {
+        impl #name {
+            #(#functions)*
+        }
+    }
+}
+
+pub fn is_context(attr: &Attribute) -> bool {
+    if let Meta::Path(path) = &attr.meta {
+        if let Some(segment) = path.segments.last() {
+            let ident = segment.ident.to_string();
+            if ident == "context" {
+                return true;
+            }
+        }
+    }
+    false
```

Comment:
```suggestion
   if let Meta::Path(path) = &attr.meta {
        return path
            .segments
            .last()
            .map_or(false, |segment| segment.ident == "context");
    }
    false
```

Reply:
I'm not sure this makes things any clearer.

Reply:
I think that what @KrosFire suggested is slightly more readable

Reply:
I've removed the unnecessary `.to_string()`, but that `.map_or()` construct seems unnecessarily verbose.